lib/puppet/provider/service/smf.rb

```diff
@@ -80,7 +80,7 @@ def startcmd
     when :maintenance, :degraded
       [command(:adm), :clear, @resource[:name]]
     else
-      [command(:adm), :enable, "-s", @resource[:name]]
+      [command(:adm), :enable, "-rs", @resource[:name]]
```

Comment:
We committed the original behavior and it's unlikely that we will ever change it. We leave it up to the user to choose the behavior they want when manually executing. However, we think it should be the default behavior for puppet.